### PR TITLE
New package: bmap-tools-3.5

### PR DESCRIPTION
**Pull request by raedwulf**

### CODE REVIEW

**Comment by raedwulf** [comment]
This python program allows you to write sparse disk images to disk without having to write those sparse/empty blocks.

File: srcpkgs/bmap-tools/template [hunk 1]
```diff
@@ -0,0 +1,19 @@
+# Template file for 'bmap-tools'
+pkgname=bmap-tools
+version=3.5
+revision=1
+archs=noarch
+build_style=python3-module
+make_install_args="--optimize=1"
+hostmakedepends="python3-setuptools"
+depends="python3-setuptools python3-six"
```
Comment (abenson) [inline srcpkgs/bmap-tools/template:9]: It needs setuptools at runtime?

Comment (raedwulf) [reply srcpkgs/bmap-tools/template]: Good catch, thanks! :+1:

**Comment by Chocimier** [comment]
Not compatible with python 3.8:

```
ERROR: test (tests.test_api_base.TestCreateCopy)
The test entry point. Executes the '_do_test()' function for files of
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/builddir/bmap-tools-3.5/bmaptools/Filemap.py", line 479, in get_mapped_ranges
    first_prev, last_prev = next(iterator)
StopIteration

```


It _depends_ on setuptools:
```
tar xOf hostdir/binpkgs/bmap-tools-3.5_1.noarch.xbps ./usr/bin/bmaptool   
#!/usr/bin/python3
# EASY-INSTALL-ENTRY-SCRIPT: 'bmap-tools==3.5','console_scripts','bmaptool'
__requires__ = 'bmap-tools==3.5'
import re
import sys
from pkg_resources import load_entry_point

if __name__ == '__main__':
    sys.argv[0] = re.sub(r'(-script\.pyw?|\.exe)?$', '', sys.argv[0])
    sys.exit(
        load_entry_point('bmap-tools==3.5', 'console_scripts', 'bmaptool')()
    )
```

Also duplicate of #20583, but improved a bit.

cc @anjandev

**Comment by raedwulf** [comment]
> Not compatible with python 3.8:

I added all the upstream patches since 3.5 since they are all useful; but I don't know if I should remove the ones that add new features (like the zstd one) and only keep bugfixes.

> It depends on setuptools:

I added it back in, thanks - I thought I misremembered that it weirdly needed it.

File: srcpkgs/bmap-tools/template [hunk 2]
```diff
@@ -0,0 +1,20 @@
+# Template file for 'bmap-tools'
+pkgname=bmap-tools
+version=3.5
+revision=1
+archs=noarch
+build_style=python3-module
+make_install_args="--optimize=1"
```
Comment (Chocimier) [inline srcpkgs/bmap-tools/template:7]: --optimize is not necessary. Xbps don't use cache created duiring build, it creates one on install, and it is optimized :)

**Comment by Chocimier** [comment]
We patch as few as necessary, please remove patch of tests ond one for zstd. Otherwise looks good.

File: srcpkgs/bmap-tools/template [hunk 3]
```diff
@@ -0,0 +1,20 @@
...
+hostmakedepends="python3-setuptools"
+depends="python3-setuptools python3-six"
```
Comment (Chocimier) [inline srcpkgs/bmap-tools/template:9]: please add checkdepends="${depends}"

File: srcpkgs/bmap-tools/template (hunk 2 above)
Comment (raedwulf) [reply srcpkgs/bmap-tools/template]: Good to know, thanks!

File: srcpkgs/bmap-tools/template (hunk 3 above)
Comment (raedwulf) [reply srcpkgs/bmap-tools/template:8]: I've made the change, though I'm curious as to what this does.

Comment (Chocimier) [reply srcpkgs/bmap-tools/template:8]: Xbps-src can run test suite of packaged software. It's run with `check` target or when `XBPS_CHECK_PKGS=yes` is set in etc/conf. Without dependencies, tests failed.

Comment (raedwulf) [reply srcpkgs/bmap-tools/template:8]: Thanks :+1:

**Pull request closed without merging**